### DISCUSSION TITLE
How to open hide with shortcut

### DISCUSSION THREAD

**Discussion by woile** [wezterm/wezterm#1408] [Q&A]
Hello, I'm new to wezterm and terminals in general, I usually have `guake` binded to `F10`, which either opens a new or already opened tabs; or hides the existing terminals.

How can I achieve the same with wezterm?

Thanks!

**Comment by wez** [comment]
wezterm doesn't have support for this workflow today, but could grow it in the future.

**Reply by woile** [reply]
Thanks! I'd love to have a dedicated key for my terminal, and wezterm works super good so far!

**Comment by naquad** [comment]
Necroposting all the way...

WeztermDD = Wezterm + Devilspie + xDotool

[Devilspie](https://www.burtonini.com/blog/tag/devilspie.html) from Arch Linux repo, version 0.23.
Autostarted, keeps an eye on the Wezterm running with `--class=weztermdd`:

```scheme
(begin
  (if (is (window_class) "weztermdd") (begin
    (geometry "3072x1080+384+2160") ; CHANGE THIS ACCORDINGLY
    (skip_tasklist)
    (skip_pager)
    (stick)
  ))
)
```

Wrapper script
```bash
#!/bin/sh

export WEZTERMDD=1

xdotool search --onlyvisible --class weztermdd windowunmap || \
  xdotool search --class weztermdd windowmap || \
  exec wezterm start --always-new-process --class=weztermdd
```

Wezterm configuration section responsible for dropdown handling:

```lua
local is_dropdown = os.getenv('WEZTERMDD') == '1'
local keys = {
  -- your usual key configuration
}

if is_dropdown then
  wezterm.on('window-focus-changed', function(window)
    if not window:is_focused() then
      -- when the window loses the focus, it should be hidden just like other drop-down apps
      wezterm.background_child_process(
        wezterm.shell_split('xdotool search --class weztermdd windowunmap')
      )
    end
  end)

  config.adjust_window_size_when_changing_font_size = false
  config.tab_bar_at_bottom = true

  -- suppress new window creation to avoid adventures with multiple dropdown windows
  table.insert(keys, { key = 'N', mods = 'CTRL', action = act.Nop })
  table.insert(keys, { key = 'n', mods = 'SUPER', action = act.Nop })
  table.insert(keys, { key = 'N', mods = 'SHIFT|CTRL', action = act.Nop })
  table.insert(keys, { key = 'n', mods = 'SHIFT|CTRL', action = act.Nop })
else
  -- normal configuration
end

config.keys = keys
```

Can I humbly attract your attention to this issue @wez ? We really need a remedy form all these hacks and workarounds.

**Comment by rabbitvictor** [comment]
There's also [tdrop](https://github.com/noctuid/tdrop) - really easy to configure, but only works for X Window Managers.

**Reply by naquad** [reply]
It was an obvious first thing to try but I couldn't configure it as described. If you'll manage to do so, please share the configuration.

**Reply by rabbitvictor** [reply]
Sure!

Installing tdrop:

```
#!/bin/bash

# Install dependencies first
sudo apt-get install xdotool
sudo apt-get install x11-utils
sudo apt-get install x11-xserver-utils
sudo apt-get install gawk

cd "$HOME" || exit
mkdir tdrop


#ref: https://github.com/noctuid/tdrop
git clone git@github.com:noctuid/tdrop.git
cd tdrop || exit
sudo install -D -m 755 tdrop "/usr/local/bin"/tdrop
```

Then I create a keyboard shortcut for the following tdrop command: `tdrop -a -m -h 75% wezterm`. Whenever you use the shortcut, it either brings the screen up or hides it.

 I have a script for that, but I'm sure it varies for different linux distributions.

Hope it helps!

**Reply by naquad** [reply]
It is a simple setup without the auto hide, toggling, and hide on blur.
